### DISCUSSION TITLE
Shared disk for workers. Is this possible or even a good idea?

### DISCUSSION THREAD

**Discussion by pms1969** [concourse/concourse#8251] [Help & Support] [answered]
One of our biggests costs for concourse is the network traffic that it generates.  I've been thinking of architecting the cluster so that the workers share the same disk (EFS in AWS, or a shared multiattach EBS volume), and was hoping that if they all had the same base directory, this would significantly reduce the amount of network traffic.
Before I embark on such a hefty rework of our infra, is anyone aware of blockers to this strategy?  Is it sensible, or complete insanity?
Thanks.

**Comment by taylorsilva** [comment, marked as answer]
Copying my answer from Discord

---

I don't think you'll get this to work because of the assumptions Concourse makes about its workers.

Concourse assumes each worker has its own set of compute resources that are not shared with other workers.
With that assumption, Concourse assumes the only assured path of getting your artifacts from one worker to another would be over the network.

The current path an artifact takes is 
```
Worker -> Web node -> Worker
```

Concourse does not assume that worker disks would ever be shared between workers. There is no code supporting or even hinting that this assumption is possible.

Not sure if this will save you bandwidth, but if all your workers are on the same LAN then you could try doing P2P volume  streaming https://concourse-ci.org/concourse-worker.html#configuring-peer-to-peer-volume-streaming

**Reply by pms1969** [reply]
Thanks @taylorsilva.  We've enabled P2P now, and its helped a bit.  next step to right size the nodes.
